.buildkite/scripts/generate_pipeline.sh

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+echo "--- Install dependencies"
```

Comment:
Using the images of our team that already have Python and packages, we don't need to do anything, we just run the python scripts

Reply:
That's awesome.

OTOH, if there is no need to use external dependencies, then the `catalog-info.yaml` could invoke the Python script directly.

```yaml
...
  implementation:
    apiVersion: "buildkite.elastic.dev/v1"
    kind: "Pipeline"
    metadata:
       ...
    spec:
      pipeline_file: .buildkite/generator.py
...
```

I've seen that behaviour in some other projects. I think these projects used `json` , and IIUC, that's a standard library in python.

Buildkite supports json and yaml syntax.

IIUC, using the catalog-info.yaml approach will remove the need to git clone to load the pipeline.yml file.

Reply:
I have seen something similar, but I have not tested it.